### PR TITLE
chore: add beta release workflow action

### PR DESCRIPTION
**Pull request by rubencarvalho**
## Description
This PR introduces a new GitHub Actions workflow to automate the publishing of beta releases whenever changes are merged into the `main` branch. The beta releases are configured to always point to the next minor version.

## Related issue(s)

- [SWC-489](https://jira.corp.adobe.com/browse/SWC-489)

## Motivation and context

We're aiming to shorten feedback loops with teams, allowing consumers to test upcoming features and fixes before the official release.
This also moves us closer to a full CI/CD pipeline and will allow us to play around the idea of dry-run builds for upcoming releases.

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [x] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: .github/workflows/beta-release.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+              with:
+                  node-version: '20'
+
+            - name: Install dependencies
+              run: yarn install --frozen-lockfile
+
+            - name: Set Git identity
+              run: |
+                  git config --global user.email "rcarvalho@adobe.com"
+                  git config --global user.name "Ruben Carvalho"
```
Comment (rubencarvalho) [inline .github/workflows/beta-release.yml:27]: We will create a team Git identity.

File: projects/documentation/content/support-and-compatibility.md [hunk 2]
```diff
@@ -13,6 +13,20 @@ This page provides comprehensive information on versioning, public APIs, browser
 
 Starting from version 1.0.0, Spectrum Web Components follows semantic versioning ([semver](https://semver.org/)). We regularly release patch versions, which do not contain breaking changes. When a breaking change occurs, it will be done in a major version release to avoid breaking existing applications depending on the old version. Major version releases will be communicated in advance, and migration guides will be provided.
 
+### Beta versions
+
+To provide early access to upcoming releases, we maintain a beta tag that points to the next minor version. The beta tag will always correspond to the next minor version incremented from the latest stable release. For example, if the latest tagged release is `1.2.1`, the beta tag will be `1.3.0-beta.0`.
```
Comment (Rajdeepc) [inline projects/documentation/content/support-and-compatibility.md:18]: Is there a way we can spin up a parallel beta site with the existing netlify account?

Comment (rubencarvalho) [reply projects/documentation/content/support-and-compatibility.md:18]: I added the functionality to trigger a beta website deployment on `main`.

Here is the link: https://beta--spectrum-web-components.netlify.app/

File: .github/workflows/beta-release.yml (hunk 1 above)
Comment (rubencarvalho) [reply .github/workflows/beta-release.yml]: (for now I am using the github-actions-bot / support+actions@github.com)

File: .github/workflows/beta-release.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+                  echo "beta_version=$BETA_VERSION" >> $GITHUB_OUTPUT
+
+            - name: Update package versions for beta release
+              run: |
+                  npx lerna version "${{ steps.calculate_next_beta_version.outputs.beta_version }}" --no-git-tag-version --no-push --yes
+
+            - name: Configure NPM for Lerna publish
+              env:
+                  NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```
[the remark this answers is not in the document]
Comment (rubencarvalho) [reply .github/workflows/beta-release.yml:75]: (but let this be non-blocking)

**Comment by TarunAdobe** [comment]
code coverage failed on chromium 😔 @rubencarvalho

**Review by blunteshwar — APPROVED** [review]

**Pull request merged**